src/main/java/MyArrayList.java

```diff
@@ -0,0 +1,43 @@
+public class MyArrayList<T> {
+    private T[] array;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyArrayList() {
+        this.array = (T[]) new Object[10];
+        this.size = 0;
+    }
+
+    public void add(T element) {
+        if (size == array.length) {
+            resizeArray();
+        }
+        array[size++] = element;
+    }
+
+    public T getByIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
```

Comment:
could be one extra method. You use it twice

Reply:
you mean extract  method?

Reply:
yes)